internal/controlplane/handlers_authz.go

```diff
@@ -292,32 +296,157 @@ func (s *Server) ListRoleAssignments(
+	authzRole, err := authz.ParseRole(role)
 	if err != nil {
 		return nil, util.UserVisibleError(codes.InvalidArgument, err.Error())
 	}
 
+	// Ensure the target project exists
+	_, err = s.store.GetProjectByID(ctx, targetProject)
+	if err != nil {
+		// If the project is not found, return an error
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, util.UserVisibleError(codes.InvalidArgument, "target project with ID %s not found", targetProject)
+		}
+		return nil, status.Errorf(codes.Internal, "error getting project: %v", err)
+	}
+
+	// Validate the subject and email - decide if it's an invitation or a role assignment
+	if sub == "" && email != "" && isEmail(email) {
+		if flags.Bool(ctx, s.featureFlags, flags.UserManagement) {
+			return s.inviteUser(ctx, targetProject, authzRole, email)
+		}
+		return nil, util.UserVisibleError(codes.Unimplemented, "user management is not enabled")
+	} else if sub != "" && email == "" {
+		return s.assignRole(ctx, targetProject, authzRole, sub)
+	}
+	return nil, util.UserVisibleError(codes.InvalidArgument, "one of subject or email must be specified")
+}
+
+func (s *Server) inviteUser(
+	ctx context.Context,
+	targetProject uuid.UUID,
+	role authz.Role,
+	email string,
+) (*minder.AssignRoleResponse, error) {
+	var userInvite db.UserInvite
+	// Current user is always authorized to get themselves
+	tokenString, err := gauth.AuthFromMD(ctx, "bearer")
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "no auth token: %v", err)
+	}
+
+	openIdToken, err := s.jwt.ParseAndValidate(tokenString)
+	if err != nil {
+		return nil, status.Errorf(codes.Unauthenticated, "failed to parse bearer token: %v", err)
+	}
+
+	// Get the sponsor's user information (current user)
+	currentUser, err := s.store.GetUserBySubject(ctx, openIdToken.Subject())
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get user: %s", err)
+	}
+
+	// Check if the user is already invited
+	existingInvite, err := s.store.GetInvitationByEmailAndProjectAndRole(ctx, db.GetInvitationByEmailAndProjectAndRoleParams{
+		Email:   email,
+		Project: targetProject,
+		Role:    role.String(),
+	})
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			// If there are no invitations for this email, great, we should create one
+			userInvite, err = s.store.CreateInvitation(ctx, db.CreateInvitationParams{
+				Code:    invite.GenerateCode(),
+				Email:   email,
+				Role:    role.String(),
+				Project: targetProject,
+				Sponsor: currentUser.ID,
+			})
+			if err != nil {
+				return nil, status.Errorf(codes.Internal, "error creating invitation: %v", err)
+			}
+		} else {
+			// Some other error happened, return it
+			return nil, status.Errorf(codes.Internal, "error getting invitations: %v", err)
+		}
+	} else {
+		// If we didn't get an error, this means there's an existing invite.
+		// We should update its expiration and send the response.
+		userInvite, err = s.store.UpdateInvitation(ctx, existingInvite.Code)
```

Comment:
Here we should check the invite expiration date before determining if we should update it. We want to avoid replaying an invite in case someone gets access to an old email for example

Reply:
Let me give some context. So the lifespan of the invites should be 7 days for expiration, but we'll keep them for 30 days before deleting them entirely so an admin can update them once they see they have expired. This bumps their expires time only so previous send links continue to work and this is the intended behavior.

This flow happens when an administrator invites someone else again and there was an existing invitation (meaning it is less than 30 days old, otherwise it should have been deleted).

The scenarios are:
* If the invite expires <= 7 days old, they are fine and can accept.
* If the invite is 7 days < expires < 30 days, the invite will not work unless the admin went and updated it through this flow in the last 7 days
* If the invite is 30 < expires, it shouldn't be present in the db, that is for both the admin and the invitee.

In that sense, I don't think we need to check the expiration because this flow is called by the administrator explicitly and is about updating the expiration of an invite.